fdbserver/DataDistribution.actor.cpp

```diff
@@ -2932,11 +2945,13 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 				}
 
 				lastZeroHealthy = self->zeroHealthyTeams->get(); //set this again in case it changed from this teams health changing
-				if( self->initialFailureReactionDelay.isReady() && !self->zeroHealthyTeams->get() ) {
+				if ((self->initialFailureReactionDelay.isReady() && !self->zeroHealthyTeams->get()) ||
+				    teamContainsFailedServer(self, team)) {
 					vector<KeyRange> shards = self->shardsAffectedByTeamFailure->getShardsFor( ShardsAffectedByTeamFailure::Team(team->getServerIDs(), self->primary) );
 
 					for(int i=0; i<shards.size(); i++) {
-						int maxPriority = team->getPriority();
+						// Make it high priority to move keys off failed server or else RelocateShards may never be addressed
+						int maxPriority = teamContainsFailedServer(self, team) ? PRIORITY_TEAM_0_LEFT : team->getPriority();
```

Comment:
Replace `teamContainsFailedServer` with variable

Reply:
About the priority, 
Simply making it PRIORITY_TEAM_0_LEFT priority may block the RelocateShards that is actually PRIORITY_TEAM_0_LEFT priority.

Should we move keys based on the number of servers left in the team? For example, if there is 1 server left after marking the SS as failed, we should use PRIORITY_TEAM_1_LEFT instead of PRIORITY_TEAM_0_LEFT. Can you try this and test in correctness to see if it may cause any problem?

If we want the failed SS case to have a higher priority, we can create a new priority value in the PRIORITY_TEAM_x_LEFT band. But I don't think it can really solve any functionality problem if there is.